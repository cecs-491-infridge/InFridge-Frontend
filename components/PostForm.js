import React from 'react';
import { Text, TextInput } from 'react-native';
import { Button, Container, Content, Header, Item, Textarea, Form } from 'native-base';

class PostForm extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            body: '',
            location: '',
            tradeType: '',
            error: ''
        };
    }

    onSubmit = (e) => {
        e.preventDefault();

        const error =
            !this.state.body ? 'Please include a Post body' :
                !this.validateTradeType() ? 'Please include a valid Trade Type (donate/trade/sell)' : '';

        this.setState({ error });

        if (!error) {
            const transaction = {
                body: this.state.body,
                location: '123',
                tradeType: this.state.tradeType
            };

            this.setState({
                body: '',
                tradeType: ''
            });
            this.props.onSubmit(transaction);
        }
    }

    validateTradeType = () => {
        const tradeType = this.state.tradeType;

        if (tradeType === 'donate' || tradeType === 'trade' || tradeType === 'sell') {
            return true;
        }
        return false;
    }

    render() {
        return (
            <Content padder>

                {!!this.state.error && <Text>{this.state.error}</Text>}
                <Form>
                    <Textarea
                        bordered placeholder="Trade Type"
                        onChangeText={(tradeType) => this.setState({ tradeType })}
                        value={this.state.tradeType}
                    />


                    <Textarea
                        rowSpan={5}
                        bordered placeholder="Post body"
                        onChangeText={(body) => this.setState({ body })}
                        value={this.state.body}
                    />

                    <Button
                        block
                        onPress={this.onSubmit}
                    >

                        <Text>Create Post</Text>
                    </Button>

                </Form>






            </Content>
        )
    }
}

export default PostForm;