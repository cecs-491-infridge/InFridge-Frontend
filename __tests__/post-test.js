import React from 'react';
import PostForm from '../components/PostForm.js';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<PostForm />).toJSON();
  expect(tree).toMatchSnapshot();

    console.log(tree.validateStatusPost())
});