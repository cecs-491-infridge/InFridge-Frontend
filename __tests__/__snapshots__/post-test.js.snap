// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentContainerStyle={
    Array [
      Object {
        "padding": 10,
      },
      undefined,
    ]
  }
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutomaticScroll={true}
  enableOnAndroid={false}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  getScrollResponder={[Function]}
  handleOnScroll={[Function]}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  keyboardShouldPersistTaps="handled"
  keyboardSpace={0}
  onScroll={[Function]}
  padder={true}
  resetKeyboardSpace={[Function]}
  resetScrollToCoords={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  scrollEventThrottle={1}
  scrollForExtraHeightOnAndroid={[Function]}
  scrollIntoView={[Function]}
  scrollToEnd={[Function]}
  scrollToFocusedInput={[Function]}
  scrollToPosition={[Function]}
  showsVerticalScrollIndicator={true}
  style={
    Object {
      "backgroundColor": "transparent",
      "flex": 1,
    }
  }
  update={[Function]}
  viewIsInsideTabBar={false}
>
  <View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderBottomWidth": 0.5,
            "borderColor": "#c9c9c9",
            "flexDirection": "row",
            "marginLeft": 16,
            "paddingBottom": 13,
            "paddingRight": 16,
            "paddingTop": 13,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "alignSelf": "center",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "flex-start",
                "backgroundColor": "transparent",
                "borderBottomWidth": null,
                "borderColor": "#007aff",
                "borderLeftWidth": null,
                "borderRadius": 5,
                "borderRightWidth": null,
                "borderTopWidth": null,
                "borderWidth": undefined,
                "elevation": 0,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "space-between",
                "opacity": 1,
                "paddingBottom": 6,
                "paddingTop": 6,
                "shadowColor": null,
                "shadowOffset": null,
                "shadowOpacity": null,
                "shadowRadius": null,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 30,
                    },
                    Object {
                      "color": "#007aff",
                      "fontSize": 24,
                      "marginLeft": 16,
                      "marginRight": 16,
                      "paddingTop": 2,
                    },
                  ],
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              Ôèè
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "alignSelf": "center",
              "flex": 0.28,
              "paddingBottom": null,
              "paddingLeft": null,
              "paddingRight": null,
              "paddingTop": null,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": null,
                "backgroundColor": "#007aff",
                "borderBottomWidth": null,
                "borderColor": "#007aff",
                "borderLeftWidth": null,
                "borderRadius": 5,
                "borderRightWidth": null,
                "borderTopWidth": null,
                "borderWidth": undefined,
                "elevation": 2,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "space-between",
                "opacity": 1,
                "paddingBottom": 6,
                "paddingTop": 6,
                "shadowColor": undefined,
                "shadowOffset": undefined,
                "shadowOpacity": undefined,
                "shadowRadius": undefined,
                "width": 150,
              }
            }
          >
            <Text
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#fff",
                  "fontFamily": "System",
                  "fontSize": 16.5,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                }
              }
              uppercase={false}
            >
              Status Update
            </Text>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": null,
                "backgroundColor": "#007aff",
                "borderBottomWidth": null,
                "borderColor": "#007aff",
                "borderLeftWidth": null,
                "borderRadius": 5,
                "borderRightWidth": null,
                "borderTopWidth": null,
                "borderWidth": undefined,
                "elevation": 2,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "space-between",
                "opacity": 1,
                "paddingBottom": 6,
                "paddingTop": 6,
                "shadowColor": undefined,
                "shadowOffset": undefined,
                "shadowOpacity": undefined,
                "shadowRadius": undefined,
                "width": 150,
              }
            }
          >
            <Text
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#fff",
                  "fontFamily": "System",
                  "fontSize": 16.5,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                }
              }
              uppercase={false}
            >
              Transaction
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      bordered={true}
      style={Object {}}
    >
      <TextInput
        allowFontScaling={true}
        bordered={true}
        editable={true}
        multiline={true}
        onChangeText={[Function]}
        placeholder="Trade Type"
        placeholderTextColor="#575757"
        rejectResponderTermination={true}
        rowSpan={2}
        style={
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#D9D5DC",
            "borderLeftWidth": 1,
            "borderRightWidth": 1,
            "borderTopWidth": 1,
            "color": "#000",
            "fontSize": 15,
            "height": 50,
            "marginTop": 5,
            "paddingLeft": 10,
            "paddingRight": 5,
            "textAlignVertical": "top",
          }
        }
        underlineColorAndroid="rgba(0,0,0,0)"
        value=""
      />
      <TextInput
        allowFontScaling={true}
        bordered={true}
        editable={true}
        multiline={true}
        onChangeText={[Function]}
        placeholder="Post body"
        placeholderTextColor="#575757"
        rejectResponderTermination={true}
        rowSpan={2}
        style={
          Object {
            "borderBottomWidth": 1,
            "borderColor": "#D9D5DC",
            "borderLeftWidth": 1,
            "borderRightWidth": 1,
            "borderTopWidth": 1,
            "color": "#000",
            "fontSize": 15,
            "height": 50,
            "marginTop": 5,
            "paddingLeft": 10,
            "paddingRight": 5,
            "textAlignVertical": "top",
          }
        }
        underlineColorAndroid="rgba(0,0,0,0)"
        value=""
      />
      <View
        style={
          Array [
            Object {},
            Object {
              "alignItems": "center",
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {},
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-evenly",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {},
            Object {},
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "flex-start",
              "backgroundColor": "#007aff",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 2,
              "flexDirection": "row",
              "height": 30,
              "justifyContent": "space-between",
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
              "shadowColor": undefined,
              "shadowOffset": undefined,
              "shadowOpacity": undefined,
              "shadowRadius": undefined,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#fff",
                    "fontSize": 20,
                    "marginLeft": 16,
                    "marginRight": 16,
                    "paddingTop": 0,
                  },
                ],
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            ÔÜú
          </Text>
          <Text
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#fff",
                "fontFamily": "System",
                "fontSize": 14,
                "marginLeft": 0,
                "marginRight": 0,
                "paddingLeft": 16,
                "paddingRight": 16,
              }
            }
            uppercase={false}
          >
            Pictures
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {},
            Object {},
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "flex-start",
              "backgroundColor": "#007aff",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 2,
              "flexDirection": "row",
              "height": 30,
              "justifyContent": "space-between",
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
              "shadowColor": undefined,
              "shadowOffset": undefined,
              "shadowOpacity": undefined,
              "shadowRadius": undefined,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#fff",
                    "fontSize": 20,
                    "marginLeft": 16,
                    "marginRight": 16,
                    "paddingTop": 0,
                  },
                ],
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            Ôãõ
          </Text>
          <Text
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#fff",
                "fontFamily": "System",
                "fontSize": 14,
                "marginLeft": 0,
                "marginRight": 0,
                "paddingLeft": 16,
                "paddingRight": 16,
              }
            }
            uppercase={false}
          >
            Post
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
